rust/module.rs

```diff
@@ -155,12 +155,28 @@ fn build_modinfo_string_param(module: &str, field: &str, param: &str, content: &
         + &__build_modinfo_string_base(module, field, &content, &variable, false)
 }
 
+fn permissions_are_readonly(perms: &str) -> bool {
+    let (radix, digits) = if let Some(n) = perms.strip_prefix("0x") {
+        (16, n)
+    } else if let Some(n) = perms.strip_prefix("0o") {
+        (8, n)
+    } else if let Some(n) = perms.strip_prefix("0b") {
+        (2, n)
+    } else {
+        (10, perms)
+    };
+    match u32::from_str_radix(digits, radix) {
```

Comment:
I was thinking we could move this into its own "integer-literal-like parsing function", but then again I realize this is `module`, so for that to be useful we would need to create a `core`- like dependency that could be used by both `kernel` and `module`. So no need to do it.

Reply:
Yeah - I originally put it in `kernel`, but then realized we'd prefer to use it while doing the expansion.